MVVM/ViewModel/BatteryMonitor.cs

```diff
@@ -94,23 +100,94 @@ public Task AltBatteryInfoChanged(string subTopic, MqttApplicationMessage? msg)
 			  altData.VescId == Convert.ToInt32(LocalSettings.Singleton.WheelData.BackLeftDrive.Replace("0x", ""), 16)))
 			  return Task.CompletedTask;
 
+		CallDeferred("ResetVescTimer");
+
 		_currentVoltageAlt = _currentVoltageAlt * 0.9f + 0.1f * (float)altData.VoltsIn;
-		CallDeferred("ShowAltVoltage", true);
 
-		OnBatteryDataChanged.Invoke(0,(int)(_currentVoltageAlt*10),CheckForWarnings());
+		UpdateGeneralPowerInfo();
 
+		SendToHUD();
 		return Task.CompletedTask;
 	}
 
-	void ShowAltVoltage(bool show)
+	private void ResetVescTimer()
+	{
+		_vescUpdateTimer.SetWaitTime(LocalSettings.Singleton.Battery.ExpectedMessageInterval);
+		_vescUpdateTimer.Start();
+	}
+
+	private void NoVescDataHandler()
+	{
+		_currentVoltageAlt = 0;
+		_vescVoltageLabel.SetText("VESC Voltage:\nNoData");
+		SendToHUD();
+	}
+
+	private void UpdateGeneralPowerInfo(MqttClasses.HotswapStatus newHotswapStatus)
+	{
+		battery[0].ShowHotswapStatusHandler(newHotswapStatus.HasFlag(MqttClasses.HotswapStatus.Hotswap1));
+		battery[1].ShowHotswapStatusHandler(newHotswapStatus.HasFlag(MqttClasses.HotswapStatus.Hotswap2));
+		battery[2].ShowHotswapStatusHandler(newHotswapStatus.HasFlag(MqttClasses.HotswapStatus.Hotswap3));
+		CallDeferred("ShowInQuickData", 
+			CalculateBatteryAverageVoltage(),
+			CalculateBatterySumCurrent(),
+			newHotswapStatus.HasFlag(MqttClasses.HotswapStatus.BlackMushroom),
+			(int)(newHotswapStatus & (MqttClasses.HotswapStatus.GPIO1 | MqttClasses.HotswapStatus.GPIO2 |
+									  MqttClasses.HotswapStatus.GPIO3 | MqttClasses.HotswapStatus.GPIO4)) >> 4
+			);
+		SetMushroomState?.Invoke((newHotswapStatus.HasFlag(MqttClasses.HotswapStatus.BlackMushroom)? MqttClasses.MushroomStatus.Molded : MqttClasses.MushroomStatus.Unmolded));
+	}
+
+	private void UpdateGeneralPowerInfo()
+	{
+		CallDeferred("ShowInQuickData");
```

Comment:
Prefer not to use string here

```suggestion
		CallDeferred(nameof(ShowInQuickData));
```